lawlibrary/views/legislation.py

```diff
@@ -10,27 +9,18 @@ def get_queryset(self):
         return super().get_queryset().filter(locality=None)
 
 
-class ProvincialLegislationView(TemplateView):
-    template_name = "lawlibrary/provincial_legislation.html"
+class LocalityLegislationView(BaseLocalityLegislationView):
     navbar_link = "legislation/provincial"
 
     def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
         codes = "mp ec nc kzn gp wc lp nw fs".split()
         provinces = Locality.objects.filter(code__in=codes)
-        groups = provinces[:5], provinces[5:]
-        return super().get_context_data(province_groups=groups, **kwargs)
+        context["province_groups"] = provinces[:5], provinces[5:]
```

Comment:
When moving to the base class, change this to break the list of provinces into two groups based on the length of the locality list (which is hard-coded here).